Dataset.py

```diff
@@ -555,11 +582,15 @@ def finish_epoch(self):
     This would get called at the end of the epoch (currently optional only).
     After this, further calls to :func:`get_data` or :func:`load_seqs` are invalid,
     until a new call to :func:`init_seq_order` follows.
+
+    TODO: currently this seems to be important only for the SprintDataset
```

Comment:
Yes, so? This could potentially be important for all kinds of datasets. This potentially allows for some early cleanup, or better error handling, etc.